### ISSUE TITLE
append HTML with <a href... including & fails

### ISSUE THREAD

**Issue by inquam** [FluentDOM/FluentDOM#41]
Appending a string of html fails if it contains **<a href** with **&** in the url.
Error message given is:

**Invalid/empty content parameter.**

If i first replace the **&** to **_** or some other character the html is appended just fine.

The query is created with.

    $fd = FluentDOM::QueryCss($output, 'text/html');

The document is then extracted and passed on

    $document = $fd->getDocument();

The document is then used for the actual appending

    try {
        FluentDOM::QueryCss($document)->find('button')->parent()->after($MY_HTML_STRING);
    }
    catch(\Exception $e) {
        die($e->getMessage());
    }

A quick test can be done with something like

    FluentDOM::QueryCss($document)->find('body')->append('<a href="http://www.google.com?test1=foo&test2=bar">FooBar</a>');

which fail and 

    FluentDOM::QueryCss($document)->find('body')->append('<a href="http://www.google.com">FooBar</a>');

which works fine.

Can add that the Symphony css selector is used right now. Not sure if it's the same for the other once, but I guess  the "issue" is not with the selector but deeper into the library.

**Comment by ThomasWeinert** [comment]
It is in the standard.

```
<a href="http://www.google.com?test1=foo&test2=bar">FooBar</a>
```

Is not valid HTML. The `&` needs to be escaped as `&amp;`.

```
<a href="http://www.google.com?test1=foo&amp;test2=bar">FooBar</a>
```

`&` in HTML is used for entities, to represent special characters for example the `<` as `&lt;`.

I will see if I can improve the error message.

**Comment by inquam** [comment]
yea I read up on this and found what you refer to after posting this. I tried to fix it using HTMLTidy but that removed and fiddled with a bunch of other tags making it not suitable. I also tried with replacing using regexp etc. and although it works the performance hit is something I would like to avoid.

I do think that if fragments are treated as html instead of xml this will actually be let through.

Right now I have some code to "kind of" fix the html to get it approved. Like putting style and scripts in CDATA tags (to avoid HTML inside of it in a string for instance ot mess everything up), altering links etc.
